openquake/risklib/asset.py

```diff
@@ -545,18 +554,29 @@ def get_mmi_values(self, aggregate_by, mmi_file, exposure_hdf5):
         df = pandas.concat(dfs)
         return df[df.number > 0]
 
-    # not used yet
     def agg_by_site(self):
         """
-        :returns: an array of aggregated values indexed by site ID
+        :returns: an AssetCollection aggregated by site_id, taxonomy
         """
-        N = self['site_id'].max() + 1
+        array = numpy.sort(self.array, order=['site_id', 'taxonomy'])
+        idxs = array['site_id'] * TWO32 + array['taxonomy']
+        arrays = performance.split_array(array, idxs)
```

Comment:
Splitting assets by site also done in `assoc2()` in geo/utils.py

https://github.com/gem/oq-engine/blob/engine-3.23/openquake/hazardlib/geo/utils.py#L247

Reply:
Yes, but here I am splitting by (site_id, taxonomy) and not only by site_id.